wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/FileUtil.java

```diff
@@ -74,21 +74,17 @@ public static boolean validateFileType(final FileItemWrap newFile, final List<St
 		// If extension match is unsucessful, then move to fileMimes list
 		boolean mimeMatch = false;
 		if (!extMatch && fileMimes.size() > 0) {
-			try {
```

Comment:
You could reduce the cyclometric complexity significantly by returning true if `extMatch` is `true` before you even get in here. as the final line returns `extMatch || mimeMatch`.